gcs-proxy.go

```diff
@@ -128,6 +132,9 @@ out:
 
 func (c *DecryptGcsPayload) Response(f *proxy.Flow) {
 
+	id := uuid.New()
+	f.Request.Header.Add("x-request-id", id.String())
```

Comment:
what that be x-gcsp-response-id?  It's for response, right?

Reply:
yes, its for response, but we are the request ID from this header. Also added some details about using this header in this #discussion_r1891127509